std/functional.d

```diff
@@ -1467,3 +1467,218 @@ template forward(args...)
+        alias fas   = functionAttributes!_func;
+        alias names = ParameterIdentifierTuple!_func;
+
+        // CTFE string append generates too much grabage
+        char[] retBuff = new char[112];
+        char[] ret;
+        void append(string str)
+        {
+            const size_t newLen = (ret.length + str.length);
+            while (newLen > retBuff.length)
+                retBuff.length *= 2;
+
+            ret = retBuff[0 .. newLen];
+            ret[($ - str.length) .. $] = str;
+        }
+
+        // function attributes
+        append("pragma(inline, true) ");
+        /* Whether this function is made final or virtual should not be
+           determined by the attributes of _func. */
+        if (fas & FA.property)
+            append("@property ");
+        if (fas & FA.ref_)
+            append("ref ");
+
+        append("auto ");
+        append(arvName);
+        append("()(");
+
+        // parameter list
+        static if (names.length > 0)
+        {
+            alias types = Parameters!_func;
+            alias stcs  = ParameterStorageClassTuple!_func;
+            alias dflts = ParameterDefaults!_func;
+
+            foreach (p, name; names)
+            {
+                // storage classes
+                if ((stcs[p] & STC.lazy_) != 0)
+                    append("lazy ");
+                if ((stcs[p] & STC.scope_) != 0)
+                    append("scope ");
+                if ((stcs[p] & STC.out_) != 0)
+                    append("out ");
+
+                if ((stcs[p] & STC.return_) != 0)
+                    append("return ref ");
+                else if ((stcs[p] & STC.ref_) != 0)
+                    append("auto ref ");
+
+                // type
+                append(types[p].stringof);
+                append(" ");
+
+                // identifier
+                if (name.length > 0)
+                    append(name);
+                else
+                {
+                    append("a");
+                    append(p.stringof);
+                }
+
+                // default value
+                static if (! is(dflts[p] == void))
+                {
+                    append(" = ");
+                    append(dflts[p].stringof);
+                }
+
+                if ((p + 1) < names.length)
+                    append(", ");
+            }
+        }
+        append(") ");
+
+        // more function attributes
+        if (fas & FA.const_)
+            append("const ");
+        if (fas & FA.shared_)
+            append("shared ");
+
+        // forwarding
+        append("{ return _func(");
+        static if (names.length > 0)
+        {
+            foreach (p, name; names)
+            {
+                if (name.length > 0)
+                    append(name);
+                else
+                {
+                    append("a");
+                    append(p.stringof);
+                }
+
+                if ((p + 1) < names.length)
+                    append(", ");
```

Comment:
@d-random-contributor Thanks.

I'm not sure why I didn't do that in the first place; I've used that idiom before.